dags/nf-dynamic-challenge-dag.py

```diff
@@ -0,0 +1,61 @@
+from datetime import datetime
+
+from airflow.decorators import dag, task
+from airflow.models import Param
+
+from orca.services.nextflowtower import NextflowTowerHook
+from orca.services.nextflowtower.models import LaunchInfo
+
+
+dag_params = {
+    "tower_conn_id": Param("DYNAMIC_CHALLENGE_PROJECT_TOWER_CONN", type="string"),
+    "tower_run_name": Param("nf-dynamic-challenge-test", type="string"),
+    "tower_compute_env_type": Param("spot", type="string"),
+    "challenge_view_id": Param("syn52576179", type="string"),
+    "tower_cpus": Param("4", type="string"),
+    "tower_memory": Param("16.GB", type="string"),
+}
+
+dag_config = {
+    "schedule_interval": None,
+    "start_date": datetime(2023, 6, 1),
+    "catchup": False,
+    "default_args": {
+        "retries": 2,
+    },
+    "tags": ["nextflow_tower"],
+    "params": dag_params,
+}
+
+
+@dag(**dag_config)
+def nf_dynamic_challenge_dag():
+    @task()
+    def launch_nf_dynamic_challenge_on_tower(**context):
+        hook = NextflowTowerHook(context["params"]["tower_conn_id"])
+        info = LaunchInfo(
+            run_name=context["params"]["tower_run_name"],
+            pipeline="Sage-Bionetworks-Workflows/nf-dynamic-challenge",
+            revision="main",
+            params={
+                "view_id": context["params"]["challenge_view_id"],
+            },
+            workspace_secrets=["SYNAPSE_AUTH_TOKEN"],
+        )
+        run_id = hook.ops.launch_workflow(
+            info, context["params"]["tower_compute_env_type"]
+        )
+        return run_id
+
+    @task.sensor(poke_interval=300, timeout=604800, mode="reschedule")
+    def monitor_nf_hello_workflow(run_id: str, **context):
+        hook = NextflowTowerHook(context["params"]["tower_conn_id"])
+        workflow = hook.ops.get_workflow(run_id)
+        print(f"Current workflow state: {workflow.status.state.value}")
```

Comment:
Would it make sense to swap to using the logging package to log into at an info level? For example this is similar to format's I've used before when we ingested logs into Elasticsearch and wanted to filter logs nicely:

```
import logging
import json

# Create a logger and set log level
logger = logging.getLogger("My new logger")
logger.setLevel(logging.DEBUG)

# Create a console handler and set its log level to INFO
console_handler = logging.StreamHandler()
console_handler.setLevel(logging.DEBUG)

# Create a formatter (JSON) and add it to the console handler
formatter = logging.Formatter("%(message)s")
json_formatter = logging.Formatter(json.dumps({
    "timestamp": "%(asctime)s",
    "logger": "%(name)s",
    "level": "%(levelname)s",
    "message": "%(message)s"
}))
console_handler.setFormatter(json_formatter)

# Add the console handler to the logger to allow printing message to console
logger.addHandler(console_handler)


logger.info("My first log message")
logger.debug("My second log message")
```

Output:
```
{"timestamp": "2023-09-29 12:38:58,655", "logger": "My new logger", "level": "INFO", "message": "My first log message"}
{"timestamp": "2023-09-29 12:38:58,656", "logger": "My new logger", "level": "DEBUG", "message": "My second log message"}
```

Reply:
Ah - I think I found my answer in the logs you provided you ORCA-273:
```
[2023-09-29T18:00:01.930+0000] {logging_mixin.py:150} INFO - Current workflow state: RUNNING
```
It looks like it is already handling the pretty-print of the messages :D

Reply:
Right, the logging is mostly handled by logic in our orchestrator package, [`py-orca`](https://github.com/Sage-Bionetworks-Workflows/py-orca). IMHO, if we want to add more detailed logging it should also live there so that it is seamlessly applied to all DAGs which use that package.